CMakeLists.txt

```diff
@@ -193,6 +197,13 @@ file (GLOB p4c_extensions RELATIVE ${P4C_SOURCE_DIR}/extensions ${P4C_SOURCE_DIR
 MESSAGE ("-- Available extensions ${p4c_extensions}")
 foreach (ext ${p4c_extensions})
   if (EXISTS ${P4C_SOURCE_DIR}/extensions/${ext}/CMakeLists.txt)
+    # Generate an option that makes it possible to disable this extension.
+    string(MAKE_C_IDENTIFIER ${ext} EXT_AS_IDENTIFIER)
+    string(TOUPPER ${EXT_AS_IDENTIFIER} EXT_AS_OPTION_NAME)
+    string(CONCAT ENABLE_EXT_OPTION "ENABLE_" ${EXT_AS_OPTION_NAME})
+    string(CONCAT EXT_HELP_TEXT "Build the " ${ext} " backend")
+    OPTION (${ENABLE_EXT_OPTION} ${EXT_HELP_TEXT} ON)
+
     add_subdirectory (extensions/${ext})
```

Comment:
should this be wrapped in `if (${ENABLE_EXT_OPTION})`?

Reply:
Argh, yes, it should.